kythe/go/localrun/cmd/main.go

```diff
@@ -0,0 +1,195 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"kythe.io/kythe/go/localrun"
+	"kythe.io/kythe/go/util/datasize"
+
+	// Side effect import used to register the handler for gsutil
+	_ "kythe.io/kythe/go/storage/leveldb"
+)
+
+var (
+	languages         languageFlag = allLanguages()
+	port              int
+	workingDir        string
+	kytheRelease      string
+	outputDir         string
+	acceptedLanguages = []string{"go", "java"}
+	cacheSize         = datasize.Flag("cache_size", "3gb", "How much ram to dedicate to handling")
+)
+
+var notEnoughArgsErr = fmt.Errorf("not enough arguments")
+
+func getTargets(args []string) ([]string, error) {
+	if len(args) < 1 {
+		// TODO: In go 1.13, replace with this
+		// return []string{}, xerrors.Errorf("You provided %d arguments but expected 2: %w", )notEnoughArgsErr
+		return []string{}, notEnoughArgsErr
+	}
+	return args[:], nil
+}
+
+// usage prints out usage information for the program to stderr.
+func usage() {
+	fmt.Fprintf(os.Stderr, `localrun
+
+localrun is a small binary that can be used to build, extract, index, and serve any bazel repo
+
+Usage:
+
+localrun [--language=a,b,c] [targets+]
+
+Flag arguments:
+
+`)
+	flag.Usage()
+
+	fmt.Fprintf(os.Stderr, `
+Positional arguments:
+
+  [targets+]: A list of one or more Bazel build targets that should be included in processing.
+
+Sample invocations:
+
+Build everything:
+
+	localrun //...
+
+Build just one package:
+
+	localrun //just/this/path/...
+
+Build just go code for one path:
```

Comment:
That's correct. It might be worth considering enhancing the localrun thing to be a bit more careful about what it builds (run a `bazel query` itself), but that isn't in this version.